core/src/test/java/org/apache/iceberg/TestSnapshotJson.java

```diff
@@ -19,23 +19,24 @@
 package org.apache.iceberg;
 
 import static org.apache.iceberg.Files.localInput;
+import static org.assertj.core.api.Assertions.assertThat;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
 import java.util.List;
-import java.util.UUID;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.assertj.core.api.Assertions;
-import org.junit.Assert;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 public class TestSnapshotJson {
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir private static Path temp;
 
-  public TableOperations ops = new LocalTableOperations(temp);
+  public TableOperations ops =
+      new LocalTableOperations(java.nio.file.Files.createTempDirectory(temp, "junit"));
+
+  public TestSnapshotJson() throws IOException {}
```

Comment:
why is the constructor here?

Reply:
It doesn't need. As the above comment, I previously set `createTempDirectory` for `LocalTableOperations`, then the `createTempDirectory` needs the `IOException` handling. But as your feedback, `temp` should be set for `LocalTableOperations` and this `IOException` is not needed. Remove in the next commit.